plugins/providers/metabase/client.go

```diff
@@ -170,6 +172,134 @@ func (c *client) GetCollections() ([]*Collection, error) {
 	return collection, nil
 }
 
+func (c *client) GetGroups() ([]*Group, map[string][]map[string]interface{}, map[string][]map[string]interface{}, error) {
+
+	wg := sync.WaitGroup{}
+	wg.Add(3)
+
+	var groups []*Group
+	var err error
+	go c.fetchGroups(&wg, &groups, err)
+
+	databaseResourceGroups := make(map[string][]map[string]interface{}, 0)
+	go c.fetchDatabasePermissions(&wg, databaseResourceGroups, err)
+
+	collectionResourceGroups := make(map[string][]map[string]interface{}, 0)
+	go c.fetchCollectionPermissions(&wg, collectionResourceGroups, err)
+
+	wg.Wait()
+
+	groupMap := make(map[string]*Group, 0)
+	for _, group := range groups {
+		groupMap[fmt.Sprintf("group:%d", group.ID)] = group
+	}
+
+	addResourceToGroup(databaseResourceGroups, groupMap, "database")
+	addResourceToGroup(collectionResourceGroups, groupMap, "collection")
+
+	return groups, databaseResourceGroups, collectionResourceGroups, err
+}
+
+func (c *client) fetchGroups(wg *sync.WaitGroup, groups *[]*Group, err error) {
+	defer wg.Done()
+	req, err := c.newRequest(http.MethodGet, "/api/permissions/group", nil)
+	if err != nil {
+		return
+	}
+
+	_, err = c.do(req, &groups)
+	if err != nil {
+		return
+	}
+}
+
+func (c *client) fetchDatabasePermissions(wg *sync.WaitGroup, resourceGroups map[string][]map[string]interface{}, err error) {
+	defer wg.Done()
+
+	req, err := c.newRequest(http.MethodGet, "/api/permissions/graph", nil)
+	if err != nil {
+		return
```

Comment:
log here too. and everywhere where were are returning inside from a goroutine in error scenario.